baremetal/metal3data_manager.go

```diff
@@ -648,17 +652,17 @@ func (m *DataManager) ensureM3IPClaim(ctx context.Context, poolRef corev1.TypedL
 	if m3m == nil {
 		return reconciledClaim{m3Claim: ipClaim}, nil
 	}
-	m.Log.Info("Fetched Metal3Machine")
+	m.Log.Info("Fetched Metal3Machine", "Metal3Machine", m3m.Name)
 
 	// Fetch the BMH associated with the M3M
 	bmh, err := getHost(ctx, m3m, m.client, m.Log)
 	if err != nil {
 		return reconciledClaim{m3Claim: ipClaim}, err
 	}
 	if bmh == nil {
-		return reconciledClaim{m3Claim: ipClaim}, WithTransientError(nil, requeueAfter)
+		return reconciledClaim{m3Claim: ipClaim}, WithTransientError(errors.New("no associated BMH yet"), requeueAfter)
```

Comment:
We were already returning an error, so might as well make it a bit more descriptive...